packages/cli/commands/import/workflow.ts

```diff
@@ -93,41 +115,94 @@ export class ImportWorkflowsCommand extends Command {
 				const files = await glob(`${inputPath}/*.json`);
 				for (i = 0; i < files.length; i++) {
 					const workflow = JSON.parse(fs.readFileSync(files[i], { encoding: 'utf8' }));
-					if (credentialsEntities.length > 0) {
-						// eslint-disable-next-line
+					if (credentials.length > 0) {
 						workflow.nodes.forEach((node: INode) => {
-							this.transformCredentials(node, credentialsEntities);
+							this.transformCredentials(node, credentials);
 						});
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Workflow!.save(workflow);
+					await this.storeWorkflow(workflow, user);
 				}
 			} else {
-				const fileContents = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
+				const workflows = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
 
-				if (!Array.isArray(fileContents)) {
-					throw new Error(`File does not seem to contain workflows.`);
+				if (!Array.isArray(workflows)) {
+					throw new Error(
+						'File does not seem to contain workflows. Make sure the workflows are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
-					if (credentialsEntities.length > 0) {
-						// eslint-disable-next-line
-						fileContents[i].nodes.forEach((node: INode) => {
-							this.transformCredentials(node, credentialsEntities);
+				for (i = 0; i < workflows.length; i++) {
+					if (credentials.length > 0) {
+						workflows[i].nodes.forEach((node: INode) => {
+							this.transformCredentials(node, credentials);
 						});
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Workflow!.save(fileContents[i]);
+					await this.storeWorkflow(workflows[i], user);
 				}
```

Comment:
let's wrap this in a transaction and pass the `transactionManager` to `this.storeWorkflow` as a param e.g. `this.storeWorkflow(transactionManager, workflows[i], user)`